database/plugin/metadata/sqlite/pool.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2025 Blink Labs Software
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlite
+
+import (
+	"errors"
+
+	"github.com/blinklabs-io/dingo/database/plugin/metadata/sqlite/models"
+	"github.com/blinklabs-io/dingo/database/types"
+	"github.com/blinklabs-io/gouroboros/cbor"
+	lcommon "github.com/blinklabs-io/gouroboros/ledger/common"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+// GetPool gets a pool
+func (d *MetadataStoreSqlite) GetPool(
+	// pkh lcommon.PoolKeyHash,
+	pkh []byte,
+	txn *gorm.DB,
+) (models.Pool, error) {
+	ret := models.Pool{}
+	tmpPool := models.Pool{}
+	if txn != nil {
+		result := txn.First(&tmpPool, "pool_key_hash = ?", lcommon.Blake2b224(pkh).Bytes())
```

Comment:
I created https://github.com/blinklabs-io/gouroboros/pull/997 so we can avoid jumping through these hoops to get the hash bytes